source/Halibut/Transport/Protocol/TemporaryFileDataStreamReceiver.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.CodeDom.Compiler;
+using System.IO;
+
+namespace Halibut.Transport.Protocol
+{
+    public class TemporaryFileDataStreamReceiver : IDataStreamReceiver
+    {
+        readonly Action<Stream> writer;
+
+        public TemporaryFileDataStreamReceiver(Action<Stream> writer)
+        {
+            this.writer = writer;
+        }
+
+        public void SaveTo(string filePath)
+        {
+            using (var file = new FileStream(filePath, FileMode.Create))
+            {
+                writer(file);
+            }
+        }
+
+        public void Read(Action<Stream> reader)
+        {
+            using (var tempFiles = new TempFileCollection())
```

Comment:
I haven't used `TempFileCollection` before so this is an interesting approach. 
Elsewhere we use `Path.GetTempPath()` but if this works then im for it.
I noticed in the docs it says
> This method can be called only once for each file name extension, because it will return the same name if it is called with the same extension.

I assume this is per `TempFileCollection` and we wont get naming clashes if multiple files are transferred?